Automattic/wp-calypso
# Hide sharing modal when jetpack launchpad modal is active.

EDITED: 

In https://github.com/Automattic/jetpack/pull/33098, I fixed a bug where the launchpad modal was displayed alongside the "First post published" modal. That issue was reported here p1694530274144539-slack-C057AH42XQD. 

There is also another modal, the "Post publish sharing modal" which can also overlap with the jetpack launchpad modal.

This change hides the "Post publish sharing modal" when the jetpack launchpad modal is active.

### Implementation notes.

I found that calling `wp_add_inline_script` with `jetpack-blocks-editor` caused the script not to be rendered onto the page. That is because `jetpack-blocks-editor` isn't registered with `wp_register_script` until later in the execution cycle (in jetpack code).

I think it was wrong to ever use `jetpack-blocks-editor` so I changed it to use `wpcom-block-editor-nux-script`. This allows us to check the value before showing the "Post published sharing" modal 

https://github.com/Automattic/wp-calypso/assets/22446385/a50581bb-6707-4246-8e37-1ffe716246c0

### Testing instructions
apply this diff to your sandbox `cd apps/editing-toolkit && yarn dev --sync`
Create a new test blog site, it should have the launchpad modal active. Do not skip the launchpad modal.
Sandbox the test site and "Force reconnection" on the proxy so that the change takes affect.
Publish your first post, then go back to the editor and publish a second post.
**Expected:**
You would see the launchpad modal only
**Actual**
Without this fix, you would see both the launchapd modal, and also the "Post published sharing" modal.
